source/Engine.cpp

```diff
@@ -910,21 +910,34 @@ void Engine::Step(bool isActive)
 	}
 
 	// Draw crosshairs on any minables in range of the flagship's scanners.
-	double scanRange = flagship ? 100. * sqrt(flagship->Attributes().Get("asteroid scan power")) : 0.;
-	if(flagship && scanRange && !flagship->IsHyperspacing())
-		for(const shared_ptr<Minable> &minable : asteroids.Minables())
-		{
-			Point offset = minable->Position() - center;
-			if(offset.Length() > scanRange && flagship->GetTargetAsteroid() != minable)
-				continue;
+	if(Preferences::Has("Show minable pointers"))
+	{
+		double scanRange = flagship ? 100. * sqrt(flagship->Attributes().Get("asteroid scan power")) : 0.;
+		if(flagship && scanRange && !flagship->IsHyperspacing())
+			for(const shared_ptr<Minable> &minable : asteroids.Minables())
+			{
+				Point offset = minable->Position() - center;
+				if((offset.Length() > scanRange || flagship->GetTargetAsteroid() == minable))
+					continue;
 
-			targets.push_back({
-				offset,
-				minable->Facing(),
-				.8 * minable->Radius(),
-				GetMinablePointerColor(minable == flagship->GetTargetAsteroid()),
-				3});
-		}
+				targets.push_back({
+					offset,
+					minable->Facing(),
+					.8 * minable->Radius(),
+					GetMinablePointerColor(false),
+					3
+				});
+			}
+	}
+	const auto targetAsteroidPtr = flagship->GetTargetAsteroid();
+	if(flagship && !flagship->IsHyperspacing() && targetAsteroidPtr)
```

Comment:
```suggestion
	const auto targetAsteroidPtr = flagship ? flagship->GetTargetAsteroid() : nullptr;
	if(targetAsteroidPtr && !flagship->IsHyperspacing())
```

Also, you don't seem to be range checking this one, so the pointers persist even after the asteroid is out of my target range.

Reply:
that was the same before

Reply:
I just tested and see that, yes.